### PR TITLE
Adding documentation for supporting AKO in Public Clouds.

### PR DESCRIPTION
**Pull request by parimanur**
- `docs/AKO/public_clouds.md` Added.
- This commit moves the tenancy details from Readme to separate file.
- Fixed all the autofix warnings from the https://github.com/DavidAnson/markdownlint/blob/v0.22.0/doc/Rules.md

### CODE REVIEW

File: docs/AKO/ako_tenancy.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Tenancy support in AKO
```
Comment (sudswasavi) [inline docs/AKO/ako_tenancy.md:1]: thought I merged these changes?

Comment (sudswasavi) [reply docs/AKO/ako_tenancy.md:1]: Ah it went into a new file.

Comment (parimanur) [reply docs/AKO/ako_tenancy.md:1]: Yes.

File: docs/AKO/public_clouds.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+* AKO configures the routes for `cluster1` and `cluster2` in the **GCP** routing tables for `vpc1` and `vpc2` respectively. Since there is unique mapping for cluster and SE Group the overlapping pod cidr's routes will be in their respective VPC's.
+
+### AKO in GCP Day 0 Preparation
+
+Following lists the Day 0 preparation work required to set up AKO in GCP
+
+#### GCP Side Preparation
+
+* Kubernetes clusters hosted in GCP are already running applications connected to the backend VPC subnet.
```
Comment (apalsule) [inline docs/AKO/public_clouds.md:60]: what does this line mean., You just need Kubernetes/openshift clusters are in their own dedicated VPCs ?

Comment (sudswasavi) [inline docs/AKO/public_clouds.md:60]: Is this a preparation step or more of requirement?

File: docs/AKO/public_clouds.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+* Kubernetes and Openshift cluster node VM’s should have **IP forwarding** enabled in the [GCP VM's](https://cloud.google.com/vpc/docs/using-routes#canipforward).
+* Create a dedicated backend VPC  for each of the clusters.
+
+#### AVI side preparation
+
+* Create an GCP cloud in Avi.  Skip this step if the IaaS cloud is already created.
```
Comment (sudswasavi) [inline docs/AKO/public_clouds.md:66]: Create a

File: docs/AKO/public_clouds.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+* AKO configures the routes for `cluster1` and `cluster2` in the **Azure** routing tables for `subnet1` and `subnet2` respectively.
+
+### AKO in Azure Day 0 Preparation
+
+Following lists the Day 0 preparation work required to set up AKO in Azure
+
+#### Azure Side Preparation
+
+* Kubernetes clusters hosted in Azure are already running applications in the same VNET.
```
Comment (sudswasavi) [inline docs/AKO/public_clouds.md:93]: Is this a requirement?

File: docs/AKO/public_clouds.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+* Create a dedicated subnet for each of the clusters in **VNET1** for SE to be provisioned in.
+  * `subnet1` and `subnet2` are created for `cluster1` and `cluster2` respectively
+* Create Route Table in Azure for each subnet created above and associate it to the SE subnet.
+  * Create `RouteTable1` and `RouteTable2` and associate to `subnet1` and `subnet2` respectively.
+* Configure NSG Rules in kubernetes cluster’s subnet to allow traffic from SE Subnet.
+* Give permissions to tha AVI cloud credential to write on route tables.
+
+#### AVI side preparation
+
+* Create an GCP cloud in Avi.  Skip this step if the IaaS cloud is already created.
```
Comment (sudswasavi) [inline docs/AKO/public_clouds.md:103]: GCP cloud or azure cloud?

File: docs/AKO/public_clouds.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+#### Azure Side Preparation
+
+* Kubernetes clusters hosted in Azure are already running applications in the same VNET.
+* Create a dedicated subnet for each of the clusters in **VNET1** for SE to be provisioned in.
+  * `subnet1` and `subnet2` are created for `cluster1` and `cluster2` respectively
+* Create Route Table in Azure for each subnet created above and associate it to the SE subnet.
+  * Create `RouteTable1` and `RouteTable2` and associate to `subnet1` and `subnet2` respectively.
+* Configure NSG Rules in kubernetes cluster’s subnet to allow traffic from SE Subnet.
+* Give permissions to tha AVI cloud credential to write on route tables.
```
Comment (apalsule) [inline docs/AKO/public_clouds.md:99]: Can we define a Azure role here ?

File: docs/AKO/public_clouds.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+![Alt text](images/public_cloud/azure_routing.png?raw=true)
+
+* AKO configures the routes for `cluster1` and `cluster2` in the **Azure** routing tables for `subnet1` and `subnet2` respectively.
+
+### AKO in Azure Day 0 Preparation
+
+Following lists the Day 0 preparation work required to set up AKO in Azure
+
+#### Azure Side Preparation
```
Comment (apalsule) [inline docs/AKO/public_clouds.md:91]: ip forwarding should also be switched on here, right ?

Comment (parimanur) [reply docs/AKO/public_clouds.md:91]: For Azure Cloud its not needed.

File: docs/AKO/public_clouds.md (hunk 2 above)
Comment (parimanur) [reply docs/AKO/public_clouds.md:60]: Yes. I will remove the applications running part. Will change this to " Kubernetes / Openshift clusters running in their own dedicated VPC."

Comment (parimanur) [reply docs/AKO/public_clouds.md:60]: Right, this is requirement.

Comment (parimanur) [reply docs/AKO/public_clouds.md:60]: I am changing this to : " Kubernetes / Openshift clusters running in their own dedicated VPC."

File: docs/AKO/public_clouds.md (hunk 4 above)
Comment (parimanur) [reply docs/AKO/public_clouds.md:93]: Only VNET part is a requirement. Will rephrase and add the vnet requirement in a sub bullet. Will use " Kubernetes / Openshift clusters running in Azure." as main point.

File: docs/AKO/public_clouds.md (hunk 5 above)
Comment (parimanur) [reply docs/AKO/public_clouds.md:103]: Thanks! Will change to Azure.

File: docs/AKO/public_clouds.md (hunk 6 above)
Comment (parimanur) [reply docs/AKO/public_clouds.md:99]: Yes will link the role here.

File: docs/AKO/ako_tenancy.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+This feature allows AKO to map each kubernetes / OpenShift cluster uniquely to a tenant in Avi. `ControllerSettings.tenantsPerCluster` needs to be set to `true` to enable this feature.
+
+## Steps to enable Tenancy in AKO
+
+* Create seperate tenant for each cluster in AVI. For the below steps, lets assume `billing` tenant is created by the Avi controller admin.
+![Alt text](images/tenant_path.png?raw=true)
+* Click `create`
+![Alt text](images/new_tenant.png?raw=true)
+* Create the [`ako-admin`](roles/ako-admin.json) and[`ako-tenant`](roles/ako-tenant.json) roles which gives appropriate previliges to the ako user in `admin` and `billing` tenant.
```
Comment (hemantavi) [inline docs/AKO/ako_tenancy.md:11]: space between `and` and `[ako-tenant]`.

Comment (parimanur) [reply docs/AKO/ako_tenancy.md:11]: Addressed. Thanks!

File: docs/AKO/public_clouds.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
+# AKO in Public Clouds
+
+## AKO in GCP
+
+This section covers the support for `ClusterIP` mode for GCP IaaS cloud in Avi Controller. AKO will be managing the pod routing in the GCP.
```
[the remark this answers is not in the document]
Comment (parimanur) [reply docs/AKO/public_clouds.md:5]: I have used managing as I did not want to bring controller / SE here.

File: docs/AKO/ako_tenancy.md (hunk 8 above)
Comment (apalsule) [reply docs/AKO/ako_tenancy.md:11]: Do we run our FTs with this role ?

Comment (apalsule) [inline docs/AKO/ako_tenancy.md:11]: I checked ako-admin role seems to be wrong, it mentions NO_ACCESS for all permission types, ako-tenant mentions WRITE_ACCESS or READ_ACCESS, can you check it out.

Comment (parimanur) [reply docs/AKO/ako_tenancy.md:11]: I has very few permission with Read and Write . PERMISSION_VRFCONTEXT and PERMISSION_SERVICEENGINEGROUP being few of them.

**Review by sudswasavi — APPROVED** [review]

**Pull request merged**